### PR TITLE
Add DefaultMutualHandshakeContext* benchmarks

### PR DESCRIPTION
**Pull request by rzikm**
This PR adds microbenchmarks for mutually authenticated SslStream pair which uses SslStreamCertificateContext on the server side. This allows benchmarking TLS resume with client certificates (supported on Windows, newly supported on Linux in .NET 9 via https://github.com/dotnet/runtime/pull/102656)

### CODE REVIEW

**Review by wfurt — APPROVED** [review]

**Comment by rzikm** [comment]
CI Failures seem to be unrelated.

File: src/benchmarks/micro/libraries/System.Net.Security/SslStreamTests.Context.cs [hunk 1]
```diff
@@ -29,26 +28,34 @@ public partial class SslStreamTests
...
-                ServerCertificateContext  = _context,
+                ServerCertificateContext = _context,
             };
 
             using (var sslClient = new SslStream(client, leaveInnerStreamOpen: true, delegate { return true; }))
             using (var sslServer = new SslStream(server, leaveInnerStreamOpen: true, delegate { return true; }))
             {
                 await Task.WhenAll(
-                    sslClient.AuthenticateAsClientAsync("localhost", null, SslProtocols.None, checkCertificateRevocation: false),
+                    sslClient.AuthenticateAsClientAsync("localhost", requireClientCert ? new X509CertificateCollection() { _clientCert } : null, SslProtocols.None, checkCertificateRevocation: false),
```
Comment (adamsitnik) [inline src/benchmarks/micro/libraries/System.Net.Security/SslStreamTests.Context.cs:58]: just for my education: Why do you want to create a new instance of `X509CertificateCollection` every time the benchmark is called? Does this reflect typical real-life scenario?

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you @rzikm !

File: src/benchmarks/micro/libraries/System.Net.Security/SslStreamTests.Context.cs (hunk 1 above)
Comment (rzikm) [reply src/benchmarks/micro/libraries/System.Net.Security/SslStreamTests.Context.cs:58]: I wanted to mirror what we do in non-context benchmarks for SslStream.

https://github.com/dotnet/performance/blob/b69d1bdd939f8ccbebf62cd276d3f3d95149b1da/src/benchmarks/micro/libraries/System.Net.Security/SslStreamTests.cs#L130

Typical real-world scenarios would probably always use the overload accepting Ssl(Client|Server)AuthenticationOptions and reuse a single instance of it for multiple connections, or always create new instance entirely from scratch (as we do in linked source code).

**Comment by rzikm** [comment]
@adamsitnik I can't merge this PR because of a failed GH check on `performance-ci`, I don't think the failures are related to my changes. Can you override it and merge this?

**Comment by adamsitnik** [comment]
> Can you override it and merge this?

I don't have such permissions anymore.

But @caaavik-msft or @LoopedBard3  or @DrewScoggins should be able to do that.

**Review by LoopedBard3 — APPROVED** [review]

**Pull request merged**